pkg/agent/multicluster/mc_route_controller_test.go

```diff
@@ -119,10 +156,102 @@ var (
 			},
 		},
 	}
+
+	clusterInfoImport3 = mcv1alpha1.ClusterInfoImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-d-default-clusterinfo",
+			Namespace: "default",
+		},
+		Spec: mcv1alpha1.ClusterInfo{
+			ClusterID:   "cluster-d",
+			ServiceCIDR: "14.14.4.0/12",
+			GatewayInfos: []mcv1alpha1.GatewayInfo{
+				{
+					GatewayIP: "12.13.0.10",
+				},
+			},
+			PodCIDRs: []string{
+				"10.10.0.0/16",
+			},
+			WireGuard: &mcv1alpha1.WireGuardInfo{
+				PublicKey: "key",
+			},
+		},
+	}
+
+	_, serviceCIDR, _ = net.ParseCIDR("10.10.0.0/16")
 )
 
+func TestMCRouteControllerAsWireGuardGateway(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockInterface := routemock.NewMockInterface(ctrl)
+	networkConfig := &config.NetworkConfig{}
+	c, closeFn := newMCDefaultRouteController(t,
+		&config.NodeConfig{
+			Name: "node-4",
+			PodIPv4CIDR: &net.IPNet{
+				IP: net.ParseIP("10.10.0.0"),
+			},
+		},
+		networkConfig,
+		agent.WireGuardConfig{},
+		mockInterface,
+		TrafficEncryptionModeWireGuard,
+	)
+	defer closeFn()
+	defer c.queue.ShutDown()
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+	wireGuardNewFunc = func(nodeConfig *config.NodeConfig, wireGuardConfig *config.WireGuardConfig) (wireguard.Interface, error) {
+		return &wireguardmock.MockWireGuardClient{}, nil
+	}
+
+	finishCh := make(chan struct{})
+	go func() {
+		defer close(finishCh)
+
+		// Create Gateway4
+		c.mcClient.MulticlusterV1alpha1().Gateways(gateway4.GetNamespace()).Create(context.TODO(),
+			&gateway4, metav1.CreateOptions{})
+		c.ofClient.EXPECT().InstallMulticlusterClassifierFlows(uint32(1), true).Times(1)
+		c.processNextWorkItem()
+		c.processNextWorkItem()
+
+		// Create ClusterInfoImport3
+		c.mcClient.MulticlusterV1alpha1().ClusterInfoImports(clusterInfoImport3.GetNamespace()).
+			Create(context.TODO(), &clusterInfoImport3, metav1.CreateOptions{})
+		peerNodeIP3 := getPeerGatewayIP(clusterInfoImport3.Spec, true)
+		c.ofClient.EXPECT().InstallMulticlusterGatewayFlows(clusterInfoImport3.Name,
+			gomock.Any(), peerNodeIP3, gomock.Any(), true).Times(1)
+		_, cidr, _ := net.ParseCIDR("14.0.0.0/32")
+		mockInterface.EXPECT().AddRouteForLink(cidr, 0).Times(1)
```

Comment:
I suppose you should clean up route when the ClusterInfoImport is deleted? I didn't such logic.

Reply:
The route does not exist when testing, it was mocked. And when deleting the interface, the controller will get an error which is "not found the interface" (we do not create real interface in testing). So we don't need to clean up the route.